nexus/inventory/src/builder.rs

```diff
@@ -627,7 +627,8 @@ impl CollectionBuilder {
 
         let baseboard_id = match inventory.baseboard {
             Baseboard::Pc { .. } => None,
-            Baseboard::Gimlet { identifier, model, revision: _ } => {
+            Baseboard::Gimlet { identifier, model, .. }
+            | Baseboard::Cosmo { identifier, model, .. } => {
```

Comment:
The `model` already identifies gimlet vs cosmo, right? (Based on https://github.com/oxidecomputer/omicron/pull/9013#discussion_r2334270235.) Given that, I wonder if this should be something like `Baseboard::OxideSled` so we can continue to separate non-Oxide from Oxide? If we have separate enum variants, we're allowing some illegal states, I think (`Baseboard::Gimlet { model: "cosmo's model", ...}` and vice versa).

Reply:
A very good point.

Rev'ing the API proved harder than I had anticipated, so I've backed that portion of the diff out of this change, and modified the existing code to recognize Cosmo and Gimlet from the model string (based on my understanding of the format of those strings).

My intent is to iterate and roll forward a change to switch `Gimlet` to `OxideSled`, but do so in a separate change, so that we can get this one in without too much hassle.